examples/basic_qa_pipeline.py

```diff
@@ -0,0 +1,50 @@
+import logging
+logging.basicConfig(format="%(levelname)s - %(name)s -  %(message)s", level=logging.WARNING)
+logging.getLogger("haystack").setLevel(logging.INFO)
+
+from haystack.document_stores import ElasticsearchDocumentStore
+from haystack.utils import clean_wiki_text, convert_files_to_docs, fetch_archive_from_http, print_answers, launch_es
+from haystack.nodes import FARMReader, BM25Retriever
+from haystack.pipelines import ExtractiveQAPipeline
+from pprint import pprint
+
+
+def basic_qa_pipeline():
+    # launch and create DocumentStore
+    launch_es()
+    document_store = ElasticsearchDocumentStore(host="localhost", username="", password="", index="document")
+
+    # fetch, pre-process and write documents
+    doc_dir = "data/basic_qa_pipeline"
+    s3_url = "https://s3.eu-central-1.amazonaws.com/deepset.ai-farm-qa/datasets/documents/wiki_gameofthrones_txt1.zip"
+    fetch_archive_from_http(url=s3_url, output_dir=doc_dir)
+
+    docs = convert_files_to_docs(dir_path=doc_dir, clean_func=clean_wiki_text, split_paragraphs=True)
```

Comment:
Let's use an indexing pipeline :pray: I can help writing it, but let's get rid of this `convert_files_to_docs` everywhere we can

Reply:
Are you certain? - just asking because my impression is that these are supposed to be scripts to easily copy over. So if you're saying we should have the indexing pipeline for educational purposes then we should probably mention this in tutorials. But if you think we should definitely replace the cover_files_to_docs with an indexing pipeline here too I'll get onto it :)

Reply:
I think tutorials and example scripts target two different categories.

Tutorials are for beginners. In those we really want to introduce the minimum amount of concepts that give the most results, so if we occasionally need to hide something behind a wrapper, that is a good compromise to make. They keep them on rails and simplify/abstract away classes that are (still) too flexible for them.

The example scripts instead are for people like me, or Sols I imagine, that just want an example that they can tweak in the strangest ways. I don't want wrappers there: they will keep me on the rails and prevent me from actually applying the changes I want.

What I want from the examples folder is scripts with a full, laid out pipeline that I can twist the way I want. It can be eternally long, I don't really care. For example, look at this indexing pipeline: it's a lot to type out by hand every time! Much better if I can go into the examples folder, copy-paste it, and then add `MySuperExperimentalNode` in the middle of it to see how it plays with the rest.  `TextIndexingPipeline` won't help me with that, neither will `convert_files_to_docs`.

I hope this makes sense :sweat_smile: Let's collect feedback from others if you're not convinced about this.

Reply:
Yes it does make sense!! No convincing needed. Just wanted to make sure I got on the same page and this particular part was important for me to know: "I don't want wrappers there: they will keep me on the rails and prevent me from actually applying the changes I want." 👍🏽 
I'll push some changes and re-submit for review. Many thanks!